### PR TITLE
fix: pass in separate file system for metrics

### PR DESCRIPTION
**Pull request by rahij**

### CODE REVIEW

File: spark-tpcds-benchmark-runner/src/main/java/com/palantir/spark/benchmark/metrics/BenchmarkMetrics.java [hunk 1]
```diff
@@ -78,9 +84,7 @@ public BenchmarkMetrics(
                         resolvedExperimentName + "_history", JacksonSerializer.create(BenchmarkMetric.class))
                 .createOrOpen();
         if (!this.metricsBuffer.isEmpty()) {
-            log.warn(
-                    "Found unflushed {} metrics in the buffer; attempting to flush",
-                    SafeArg.of("numMetrics", metricsBuffer.size()));
```
Comment (lwwmanning) [inline spark-tpcds-benchmark-runner/src/main/java/com/palantir/spark/benchmark/metrics/BenchmarkMetrics.java]: I think this was good to keep

File: spark-tpcds-benchmark-runner/src/main/java/com/palantir/spark/benchmark/config/HadoopConfiguration.java [hunk 2]
```diff
@@ -74,17 +92,10 @@ default Configuration toHadoopConf() {
                         filesystems().values().stream().flatMap(fsConf -> fsConf.toHadoopConf().entrySet().stream()))
                 .collectToMap()
                 .forEach(hadoopConf::set);
-        hadoopConf.set(CommonConfigurationKeys.FS_DEFAULT_NAME_KEY, getFilesystemBaseUriOrThrow(defaultFilesystem()));
+        hadoopConf.set(CommonConfigurationKeys.FS_DEFAULT_NAME_KEY, defaultFsBaseUriString());
         return hadoopConf;
     }
 
-    default String getFilesystemBaseUriOrThrow(String filesystemName) {
```
Comment (lwwmanning) [inline spark-tpcds-benchmark-runner/src/main/java/com/palantir/spark/benchmark/config/HadoopConfiguration.java:81]: nit: I think factoring this into a function was useful/clean? so would rather not revert that bit

**Review by lwwmanning — COMMENTED** [review]
couple small comments, generally lgtm

File: spark-tpcds-benchmark-runner/src/main/java/com/palantir/spark/benchmark/config/HadoopConfiguration.java (hunk 2 above)
Comment (rahij) [reply spark-tpcds-benchmark-runner/src/main/java/com/palantir/spark/benchmark/config/HadoopConfiguration.java:81]: yep got lost in the revert, added it back

**Review by lwwmanning — APPROVED** [review]

**Pull request merged**